packages/core/admin/admin/src/components/FormInputs/Date.tsx

```diff
@@ -14,21 +13,45 @@ const DateInput = forwardRef<HTMLInputElement, InputProps>(
     const { formatMessage } = useIntl();
     const field = useField(name);
     const fieldRef = useFocusInputField<HTMLInputElement>(name);
-
     const composedRefs = useComposedRefs(ref, fieldRef);
+    const [lastValidDate, setLastValidDate] = useState<Date | null>(null);
+
     const value = typeof field.value === 'string' ? new Date(field.value) : field.value;
 
+    const handleDateChange = (date: Date | undefined) => {
+      if (!date) {
+        field.onChange(name, null);
+        setLastValidDate(null);
+        return;
+      }
+
+      // Convert to UTC midnight
+      const utcDate = toUTCMidnight(date);
+      // Save as ISO string in UTC format
+      field.onChange(name, utcDate.toISOString());
+      setLastValidDate(utcDate);
+    };
+
+    // Render the DatePicker with UTC date
     return (
       <Field.Root error={field.error} name={name} hint={hint} required={required}>
         <Field.Label action={labelAction}>{label}</Field.Label>
         <DatePicker
           ref={composedRefs}
           clearLabel={formatMessage({ id: 'clearLabel', defaultMessage: 'Clear' })}
-          onChange={(date) => {
-            field.onChange(name, date ? convertLocalDateToUTCDate(date).toISOString() : null);
+          onChange={handleDateChange}
+          onClear={() => {
+            field.onChange(name, null);
+            setLastValidDate(null);
+            return;
+          }}
+          onBlur={() => {
+            // When the input is blurred, ensure that the value doesn't change unexpectedly
+            if (field.value && !value) {
+              field.onChange(name, lastValidDate?.toISOString() ?? null);
+            }
```

Comment:
>ensure that the value doesn't change unexpectedly

How would the value change unexpectedly? reading your unit test, aren't you more trying to prevent invalid dates from sticking around?

Reply:
My bad, I forgot to update the change. In fact, those tests are not as effective as I wanted them to be in preventing this bug from happening. We will write an E2E test for it before trying to get this PR merged.